### PR TITLE
Remove diagnostic analyzer service event ability

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs [hunk 1]
```diff
@@ -249,162 +236,25 @@ public async Task TestDisabledByDefaultAnalyzerEnabledWithEditorConfig(bool enab
...
-            // wait for all events to raised
-            await ((AsynchronousOperationListener)service.Listener).ExpeditedWaitAsync().ConfigureAwait(false);
-
-            // two should have been called.
-            Assert.Equal(expectedSyntax, syntax);
-            Assert.Equal(expectedSemantic, semantic);
-        }
-
-        [Fact]
-        public async Task TestOpenFileOnlyAnalyzerDiagnostics()
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs:280]: this test isn't needed anymore.  LSP handles open/closed diagnostics through document/diagnostics and workspace/diagnostics.

File: src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs [hunk 2]
```diff
@@ -857,18 +697,7 @@ void M()
...
             var text = await document.GetTextAsync();
-            service.DiagnosticsUpdated += (s, eCollection) =>
-            {
-                foreach (var e in eCollection)
-                {
-                    diagnostics.AddRange(
-                        e.Diagnostics
-                         .Where(d => d.Id == IDEDiagnosticIds.RemoveUnnecessarySuppressionDiagnosticId)
-                         .OrderBy(d => d.DataLocation.UnmappedFileSpan.GetClampedTextSpan(text)));
-                }
```
Comment (mavasani) [inline src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs:870]: Should we retain this validation for computed diagnostics?

Comment (mavasani) [reply src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs:870]: Ah, I see you are doing it below, my bad.

Comment (CyrusNajmabadi) [reply src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs:870]: that validation exists.  it just moved down to lower in the test method.

**Review by mavasani — APPROVED** [review]

**Pull request merged**